common/persistence/sql/sqlMatchingPersistence.go

```diff
@@ -530,19 +773,132 @@ func createExecution(tx *sqlx.Tx, request *persistence.CreateWorkflowExecutionRe
 		}
 	}
 	return nil
-}
\ No newline at end of file
+}
+
+func (m *sqlMatchingManager) createTransferTasks(tx *sqlx.Tx, transferTasks []persistence.Task, shardID int, domainID, workflowID, runID string) error {
+	structs := make([]struct {
+		persistence.TransferTaskInfo
+		ShardID int `db:"shard_id"`
+	}, len(transferTasks))
+
+	for i, task := range transferTasks {
+		structs[i].ShardID = shardID
+		structs[i].DomainID = domainID
+		structs[i].WorkflowID = workflowID
+		structs[i].RunID = runID
+		structs[i].TargetDomainID = domainID
+		structs[i].TargetWorkflowID = persistence.TransferTaskTransferTargetWorkflowID
+		structs[i].TargetChildWorkflowOnly = false
+		structs[i].TaskList = ""
+		structs[i].ScheduleID = 0
+
+		switch task.GetType() {
+		case persistence.TransferTaskTypeActivityTask:
+			structs[i].TargetDomainID = task.(*persistence.ActivityTask).DomainID
+			structs[i].TaskList = task.(*persistence.ActivityTask).TaskList
+			structs[i].ScheduleID = task.(*persistence.ActivityTask).ScheduleID
+
+		case persistence.TransferTaskTypeDecisionTask:
+			structs[i].TargetDomainID = task.(*persistence.DecisionTask).DomainID
+			structs[i].TaskList = task.(*persistence.DecisionTask).TaskList
+			structs[i].ScheduleID = task.(*persistence.DecisionTask).ScheduleID
+
+		case persistence.TransferTaskTypeCancelExecution:
+			structs[i].TargetDomainID = task.(*persistence.CancelExecutionTask).TargetDomainID
+			structs[i].TargetWorkflowID = task.(*persistence.CancelExecutionTask).TargetWorkflowID
+			if task.(*persistence.CancelExecutionTask).TargetRunID != "" {
+				structs[i].TargetRunID = task.(*persistence.CancelExecutionTask).TargetRunID
+			}
+			structs[i].TargetChildWorkflowOnly = task.(*persistence.CancelExecutionTask).TargetChildWorkflowOnly
+			structs[i].ScheduleID = task.(*persistence.CancelExecutionTask).InitiatedID
+
+		case persistence.TransferTaskTypeSignalExecution:
+			structs[i].TargetDomainID = task.(*persistence.SignalExecutionTask).TargetDomainID
+			structs[i].TargetWorkflowID = task.(*persistence.SignalExecutionTask).TargetWorkflowID
+			if task.(*persistence.SignalExecutionTask).TargetRunID != "" {
+				structs[i].TargetRunID = task.(*persistence.SignalExecutionTask).TargetRunID
+			}
+			structs[i].TargetChildWorkflowOnly = task.(*persistence.SignalExecutionTask).TargetChildWorkflowOnly
+			structs[i].ScheduleID = task.(*persistence.SignalExecutionTask).InitiatedID
+
+		case persistence.TransferTaskTypeStartChildExecution:
+			structs[i].TargetDomainID = task.(*persistence.StartChildExecutionTask).TargetDomainID
+			structs[i].TargetWorkflowID = task.(*persistence.StartChildExecutionTask).TargetWorkflowID
+			structs[i].ScheduleID = task.(*persistence.StartChildExecutionTask).InitiatedID
+
+		case persistence.TransferTaskTypeCloseExecution:
+			// No explicit property needs to be set
+
+		default:
+			// hmm what should i do here?
```

Comment:
return error? How does cassandra impl handle this currently?

Reply:
I'm not sure if `d.logger.Fatal` panics?

Reply:
logger.Fatal will panic inside the call.